collect_app/src/main/java/org/odk/collect/android/activities/FormFillingActivity.java

```diff
@@ -562,6 +566,7 @@ private void setupViewModels(FormEntryViewModelFactory formEntryViewModelFactory
                 swipeHandler.setBeenSwiped(false);
             } else if (validationResult instanceof SuccessValidationResult) {
                 SnackbarUtils.showLongSnackbar(findViewById(R.id.llParent), getString(org.odk.collect.strings.R.string.success_form_validation), findViewById(R.id.buttonholder));
+                ((MutableLiveData) validation).setValue(null);
```

Comment:
I'm not a fan of casting here as it breaks the encapsulation of `FormEntryViewModel` - `FormFillingActivity` is making an assumption about the return type of `getValidationResult` that the type information itself doesn't back up.

We do have a way of dealing with these "single fire events" when using `LiveData`: [`Consumable`](https://github.com/getodk/collect/blob/b6c65fc4ee5f02e282e4aa70ea3c96fae0bc3b28/androidshared/src/main/java/org/odk/collect/androidshared/data/Consumable.kt). There's an example of this in use in [`ProjectPreferencesFragment`](https://github.com/getodk/collect/blob/b6c65fc4ee5f02e282e4aa70ea3c96fae0bc3b28/collect_app/src/main/java/org/odk/collect/android/preferences/screens/ProjectPreferencesFragment.kt#L48).

Reply:
Thanks for the tip, a bit advanced for me but I got it to work after some puzzling. 

I've also tested with a dummy`FailedValidationResult`. 

And thanks @ln for the kind welcome, it's nice to be back.